product_multi_price/views/product_pricelist_views.xml

```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="product_pricelist_item_form_view" model="ir.ui.view">
+        <field name="model">product.pricelist.item</field>
+        <field name="inherit_id" ref="product.product_pricelist_item_form_view"/>
+        <field name="arch" type="xml">
+            <field name="base" position="after">
+                <field name="multi_price_name" colspan="6" attrs="{'invisible': [('base', '!=', 'multi_price')], 'required': [('base', '=', 'multi_price')]}" widget="selection"/>
```

Comment:
As @pedrobaeza said, it seems [you cannot use the selection widget in a m2o field](https://github.com/odoo/odoo/blob/7ea6798fb8b8b0249f56cf4bc5cab627438cd636/addons/web/static/src/js/fields/relational_fields.js#L2352):

```suggestion
                <field name="multi_price_name" colspan="6" attrs="{'invisible': [('base', '!=', 'multi_price')], 'required': [('base', '=', 'multi_price')]}" />
```